gateway/gateway.go

```diff
@@ -1491,6 +1494,38 @@ func (*HandleT) GetWebhookSourceDefName(writeKey string) (name string, ok bool)
 	return
 }
 
+// SaveErrors saves errors to the error db
+func (gateway *HandleT) SaveWebhookFailures(reqs []*model.FailedWebhookPayload) error {
+	jobs := make([]*jobsdb.JobT, 0, len(reqs))
+	for _, req := range reqs {
+		params := map[string]interface{}{
+			"source_id":   gateway.getSourceIDForWriteKey(req.WriteKey),
+			"stage":       "webhook",
+			"source_type": req.SourceType,
+			"reason":      req.Reason,
+		}
+		marshalledParams, err := json.Marshal(params)
+		if err != nil {
+			gateway.logger.Errorf("[Gateway] Failed to marshal parameters map. Parameters: %+v", params)
+			marshalledParams = []byte(`{"error": "rudder-server gateway failed to marshal params"}`)
+		}
+
+		jobs = append(jobs, &jobsdb.JobT{
+			UUID:         uuid.New(),
+			UserID:       uuid.New().String(), // TODO remind for a better value
+			Parameters:   marshalledParams,
+			CustomVal:    CustomVal,
```

Comment:
Does this need to be `GW` or something else? How will the current replay process know how to exclude these events in case we are replaying gateway events @cisse21?

Reply:
In general, we use both `customVal` column and `stage` field in `parameters` column to replay. 

In this case, `stage` is set to webhook and `sourceType` is set to source definition name, which can be used for replays. 

If this requires a better `customVal`, pls mention. I am open to change it.

Reply:
Let's use `WEBHOOK` or source definition name?

Reply:
@cisse21 ?

Reply:
When we replay from `proc_errors` we use a field called stage as chandu mentioned we should be good enough, also as we don't have any runbook to replay these events for now we should be good too

CustomVal in the existing replay shouldn't matter because GW replay and proc_error replay use different files altogether with different file formats and file structures but setting it as WEBHOOK makes more sense for someone looking at the backed up entries